astroquery/query.py

```diff
@@ -304,13 +318,21 @@ def _download_file(self, url, local_filepath, timeout=None, auth=None,
         else:
             length = None
 
+        stale = True  # by default consider the local copy outdated
+        if 'Date' in response.headers:
+            date = eut.parsedate(response.headers['Date'])
+            now = datetime.utcnow()
+            if (now - datetime.fromtimestamp(mktime(date))).seconds > 600:
```

Comment:
impressive wizardry. I only wonder whether this could cause any issues if the local and response times are not in sync, or `response.headers['Date']` is a local timestamp rather than something coming from the server side?